shell/platform/windows/flutter_windows_unittests.cc

```diff
@@ -90,5 +75,76 @@ TEST_F(WindowsTest, LaunchConflictingCustomEntrypoints) {
   ASSERT_FALSE(FlutterDesktopEngineRun(engine.get(), "conflictingEntrypoint"));
 }
 
+// Verify that native functions can be registered and resolved.
+TEST_F(WindowsTest, VerifyNativeFunction) {
+  auto& context = GetContext();
+  WindowsConfigBuilder builder(context);
+  builder.SetDartEntrypoint("verifyNativeFunction");
+
+  static fml::AutoResetWaitableEvent latch;
+  context.AddNativeFunction("Signal", [](Dart_NativeArguments args) {
+    latch.Signal();
+  });
+
+  ViewControllerPtr controller{builder.Run()};
+  ASSERT_NE(controller, nullptr);
+
+  // Wait until signal has been called.
+  latch.Wait();
+}
+
+// Verify that native functions that pass parameters can be registered and
+// resolved.
+TEST_F(WindowsTest, VerifyNativeFunctionWithParameters) {
+  auto& context = GetContext();
+  WindowsConfigBuilder builder(context);
+  builder.SetDartEntrypoint("verifyNativeFunctionWithParameters");
+
+  static fml::AutoResetWaitableEvent latch;
+  context.AddNativeFunction("SignalBoolValue", [](Dart_NativeArguments args) {
+    bool bool_value = false;
+    auto handle = Dart_GetNativeBooleanArgument(args, 0, &bool_value);
+    ASSERT_FALSE(Dart_IsError(handle));
+    EXPECT_TRUE(bool_value);
+    latch.Signal();
+  });
+
+  ViewControllerPtr controller{builder.Run()};
+  ASSERT_NE(controller, nullptr);
+
+  // Wait until signalBoolValue has been called.
+  latch.Wait();
+}
+
+// Verify that native functions that return values can be registered and
+// resolved.
+TEST_F(WindowsTest, VerifyNativeFunctionWithReturn) {
+  auto& context = GetContext();
+  WindowsConfigBuilder builder(context);
+  builder.SetDartEntrypoint("verifyNativeFunctionWithReturn");
+
+  static fml::AutoResetWaitableEvent latch;
```

Comment:
Nit: consider using a `CountDownLatch`

Reply:
I don't feel strongly about this, code is perfectly fine as-is if you prefer it currently 😄

Reply:
Nice call! Filed in the department of FML utility classes I'd forgotten about! Done.